src/DecoratorManifold.jl

```diff
@@ -0,0 +1,475 @@
+for the case that a decorator is the default decorator, see [`is_default_decorator`](@ref).
+"""
+function is_decorator_transparent(f, M::Manifold, args...)
+    return decorator_transparent_dispatch(f, M, args...) == Val(:transparent)
+end
+
+"""
+    decorator_transparent_dispatch(f, M, arge...)
+
+Given a [`Manifold`](@ref) `M` and a function `f(M,arge...)`, indicate, whether a
+function is `Val(:transparent)` or `Val(:intransparent)` for the (decorated)
+[`Manifold`](@ref) `M`. Another possibility is, that for `M` and given `args...`
+the function `f` should invoke `M`s `Val(:parent)` implementation.
+"""
+decorator_transparent_dispatch(f, M::Manifold, args...) = Val(:transparent)
+
+function _acts_transparently(f, M::Manifold, args...)
+    return _val_or(default_decorator_dispatch(M), decorator_transparent_dispatch(f, M, args...))
+end
+
+_val_or(::Val{true}, ::Val{T}) where {T} = Val(:transparent)
+_val_or(::Val{false}, ::Val{T}) where {T} = Val(T)
+
+#
+# Functions overwritten with decorators
+#
+
+function base_manifold(M::AbstractDecoratorManifold, depth::Val{N} = Val(-1)) where {N}
+    return (N != 0) ? base_manifold(M.manifold, (N > 0) ? Val(N-1) : depth) : M
+end
+
+@decorator_transparent_signature check_manifold_point(
+    M::AbstractDecoratorManifold,
+    p;
+    kwargs...,
+)
+
+@decorator_transparent_signature check_tangent_vector(
+    M::AbstractDecoratorManifold,
+    p,
+    X;
+    kwargs...,
+)
+
+@decorator_transparent_signature distance(
+    M::AbstractDecoratorManifold,
+    p,
+    q,
+)
+
+@decorator_transparent_signature exp!(
+    M::AbstractDecoratorManifold,
+    q,
+    p,
+    X,
+)
+
+@decorator_transparent_signature exp!(
+    M::AbstractDecoratorManifold,
+    q,
+    p,
+    X,
+    T
+)
+
+@decorator_transparent_signature injectivity_radius(M::AbstractDecoratorManifold)
+@decorator_transparent_signature injectivity_radius(M::AbstractDecoratorManifold, p)
+
+@decorator_transparent_signature inner(M::AbstractDecoratorManifold, p, X, Y)
+
+@decorator_transparent_signature inverse_retract!(
+    M::AbstractDecoratorManifold,
+    X,
+    p,
+    q,
+    m::AbstractInverseRetractionMethod,
+)
+
+@decorator_transparent_signature inverse_retract!(
+    M::AbstractDecoratorManifold,
+    X,
+    p,
+    q,
+    m::LogarithmicInverseRetraction,
+)
+
+@decorator_transparent_signature isapprox(M::AbstractDecoratorManifold, p, q; kwargs...)
+@decorator_transparent_signature isapprox(M::AbstractDecoratorManifold, p, X, Y; kwargs...)
+
+@decorator_transparent_signature log!(M::AbstractDecoratorManifold, X, p, q)
+
+@decorator_transparent_signature manifold_dimension(M::AbstractDecoratorManifold)
+
+@decorator_transparent_signature project_point!(M::AbstractDecoratorManifold, q, p)
+
+@decorator_transparent_signature project_tangent!(M::AbstractDecoratorManifold, Y, p, X)
+
+@decorator_transparent_signature projected_distribution(M::AbstractDecoratorManifold, d, p)
```

Comment:
I think we should leave distribution-related stuff in Manifolds.jl for now.

Reply:
Oh, yes; I even added them over there to distributions, but forgot to delete them here.